src/spec/integration/import.spec.coffee

```diff
@@ -141,6 +141,90 @@ describe 'Import integration test', ->
         expect(p.name).toEqual en: @newProductName
         expect(p.slug).toEqual en: @newProductSlug
         expect(p.key).toEqual 'productKey'
+        expect(p.state).toBeUndefined
+        expect(p.masterVariant.key).toEqual 'variantKey'
+        done()
+      .catch (err) -> done _.prettify(err)
+
+    it 'should set state for a newly-created product when configured to do so', (done) ->
+      csv =
+        """
+        productType,name,variantId,slug,key,variantKey,state
+        #{@productType.id},#{@newProductName},1,#{@newProductSlug},productKey,variantKey,
+        """
+      @importer = new Import {
+        authConfig: authConfig
+        httpConfig: httpConfig
+        userAgentConfig: userAgentConfig
+        defaultState: 'previous-state'
```

Comment:
Are you ensuring somewhere that this state is created before you run tests?

Reply:
Yes, `testhelpers.coffee` ll. 195ff.  Except in one test where it deliberately doesn't exist, to test it doesn't fall over if someone runs the process with a typo in the state key.

Reply:
Thanks.. I see it now - https://github.com/sphereio/sphere-node-product-csv-sync/blob/master/src/spec/integration/import.spec.coffee#L54